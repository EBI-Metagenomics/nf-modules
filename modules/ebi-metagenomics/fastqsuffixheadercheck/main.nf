
process FASTQSUFFIXHEADERCHECK {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mgnify-pipelines-toolkit:1.2.11--pyhdfd78af_0':
        'biocontainers/mgnify-pipelines-toolkit:1.2.11--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fastq)

    output:
    tuple val(meta), path("*.json"), emit: json
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def fastq_input = meta.single_end ? "-f ${fastq}" : "-f ${fastq[0]} -r ${fastq[1]}"

    """
    fastq_suffix_header_check \\
    ${fastq_input} \\
    -s ${prefix} \\
    -o ./

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    touch ${prefix}_suffix_header_err.json

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        mgnify-pipelines-toolkit: \$(get_mpt_version)
    END_VERSIONS
    """
}
