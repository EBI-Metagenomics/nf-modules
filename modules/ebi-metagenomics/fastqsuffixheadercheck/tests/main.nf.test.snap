{
    "fastqsuffixheadercheck - fail - PE": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,735222eb9c22d4d81b312a9682d55679"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,735222eb9c22d4d81b312a9682d55679"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-18T11:33:28.905264"
    },
    "fastqsuffixheadercheck - pass - PE": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-18T11:33:20.985434"
    },
    "fastqsuffixheadercheck - pass - SE": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-18T11:33:13.151863"
    },
    "fastqsuffixheadercheck - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ],
                "json": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test_suffix_header_err.json:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3c3c2c79df154c1f2b1bfffb28209a2f"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-06-18T11:33:36.574452"
    }
}